addons/project_timesheet_holidays/tests/test_timesheet_global_time_off.py

```diff
@@ -175,3 +175,42 @@ def test_timesheet_creation_and_deletion_for_calendar_set_and_remove(self):
         # The standard calendar is for 8 hours/day from 8 to 12 and from 13 to 17.
         # So we need to check that the timesheets don't have more than 8 hours per day.
         self.assertEqual(leave_task.effective_hours, 80)
+
+    @freeze_time('2021-01-04 08:00:00')
+    def test_creation_deletion_on_employee_calender_change(self):
+        leave_start_datetime = datetime(2021, 1, 4, 7, 0)
+        leave_end_datetime = datetime(2021, 1, 4, 18, 0)
+
+        # testing employee for toggling the resource_calendar_id
+        test_employee = self.env['hr.employee'].create({
+            'name': 'Eren Yeager',
+            'company_id': self.test_company.id,
+            'resource_calendar_id': self.test_company.resource_calendar_id.id,
+        })
+
+        self.env['resource.calendar.leaves'].create({
+            'name': 'Test',
+            'calendar_id': self.test_company.resource_calendar_id.id,
+            'date_from': leave_start_datetime,
+            'date_to': leave_end_datetime,
+        })
+
+        leave_task = self.test_company.leave_timesheet_task_id
+        # it returns timesheet which is created for the test_employee
+        def has_test_employee_timesheet():
+            return leave_task.timesheet_ids.filtered(lambda timesheet: timesheet.employee_id == test_employee)
+        self.assertTrue(has_test_employee_timesheet())
+
+        test_employee.resource_calendar_id = self.part_time_calendar.id
+        # the employee shouldn't have any timesheet recorded because the resource_calendar_id (Working hours) has been changed for employee
+        self.assertFalse(has_test_employee_timesheet())
```

Comment:
```suggestion
        self.assertFalse(has_test_employee_timesheet(),
            "the employee shouldn't have any timesheet recorded because the resource_calendar_id (Working hours) has been changed for employee"
        )
```
That is what I meant (for all assertions in the test)